app/drivers/kscan/kscan_gpio_direct.c

```diff
@@ -4,239 +4,325 @@
+                 "ZMK_KSCAN_DEBOUNCE_PRESS_MS or debounce-press-ms is too large");                 \
+    BUILD_ASSERT(INST_DEBOUNCE_RELEASE_MS(index) <= DEBOUNCE_COUNTER_MAX,                          \
+                 "ZMK_KSCAN_DEBOUNCE_RELEASE_MS or debounce-release-ms is too large");             \
+                                                                                                   \
+    static const struct gpio_dt_spec kscan_direct_inputs_##index[] = {                             \
+        UTIL_LISTIFY(INST_INPUTS_LEN(index), KSCAN_DIRECT_INPUT_CFG_INIT, index)};                 \
+                                                                                                   \
+    static struct debounce_state kscan_direct_state_##index[INST_INPUTS_LEN(index)];               \
+                                                                                                   \
+    COND_INTERRUPTS((static struct kscan_direct_irq_callback                                       \
+                         kscan_direct_irqs_##index[INST_INPUTS_LEN(index)];))                      \
+                                                                                                   \
+    static struct kscan_direct_data kscan_direct_data_##index = {                                  \
+        .pin_state = kscan_direct_state_##index,                                                   \
+        COND_INTERRUPTS((.irqs = kscan_direct_irqs_##index, ))};                                   \
+                                                                                                   \
+    static struct kscan_direct_config kscan_direct_config_##index = {                              \
+        .inputs = KSCAN_GPIO_LIST(kscan_direct_inputs_##index),                                    \
+        .debounce_config =                                                                         \
+            {                                                                                      \
+                .debounce_press_ms = INST_DEBOUNCE_PRESS_MS(index),                                \
+                .debounce_release_ms = INST_DEBOUNCE_RELEASE_MS(index),                            \
+            },                                                                                     \
+        .debounce_scan_period_ms = DT_INST_PROP(index, debounce_scan_period_ms),                   \
+        .poll_period_ms = DT_INST_PROP(index, poll_period_ms),                                     \
+    };                                                                                             \
+                                                                                                   \
+    DEVICE_DT_INST_DEFINE(index, &kscan_direct_init, NULL, &kscan_direct_data_##index,             \
+                          &kscan_direct_config_##index, APPLICATION,                               \
```

Comment:
Any reason you moved this to `APPLICATION` instead of `POST_KERNEL`?

Reply:
Copy/paste from the matrix driver, where it looks like it got switched from `POST_KERNEL` to `APPLICATION` in e448f2d32be5c1ccac2e015a18ed93e4e1eb7864. Do you remember why you did that there?